File: my-solutions/28. Find the Index of the First Occurrence in a String/code.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        # initialize index i
        i = 0
        # set index j to be the length of needle
        j = len(needle)

        # while j is less then or equal to the length of haystack
        while j <= len(haystack):

            # the current substring of haystack is equal to needle, return index i
            if haystack[i:j] == needle:
                return i
            
            # if the above condition is not met, increase index i and index j
            i += 1
            j += 1
        
        # return -1 when needle is not part of haystack 
        return -1

File: my-solutions/28. Find the Index of the First Occurrence in a String/code.cs
public class Solution {
    public int StrStr(string haystack, string needle) {
        int i = 0;
        int needleLength = needle.Length;

        while (i + needleLength <= haystack.Length)
        {
            if (haystack.Substring(i, needleLength) == needle)
            {
                return i;
            }
            
            i += 1;
        }
      
        return -1;
    }
}
